implementations/rust/ockam/ockam_vault/src/aws.rs

```diff
@@ -0,0 +1,270 @@
+use aws_sdk_kms::error::{CreateKeyError, GetPublicKeyError, SignError, VerifyError};
+use aws_sdk_kms::error::{ScheduleKeyDeletionError, ScheduleKeyDeletionErrorKind};
+use aws_sdk_kms::model::{KeySpec, KeyUsageType, MessageType, SigningAlgorithmSpec};
+use aws_sdk_kms::types::{Blob, SdkError};
+use aws_sdk_kms::Client;
+use ockam_core::vault::SecretType;
+use ockam_core::vault::{KeyId, PublicKey, Signature};
+use ockam_core::Result;
+use sha2::{Digest, Sha256};
+use thiserror::Error;
+use tracing as log;
+
+/// AWS KMS configuration.
+#[derive(Debug, Default, Clone)]
+pub struct Config {
+    multi_region: bool,
+}
+
+impl Config {
+    /// Create multi-region keys.
+    pub fn multi_region(mut self, val: bool) -> Self {
+        self.multi_region = val;
+        self
+    }
+}
+
+/// AWS KMS client.
+#[derive(Debug, Clone)]
+pub struct Kms {
+    client: Client,
+    config: Config,
+}
+
+impl Kms {
+    /// Create a new AWS KMS client.
+    pub async fn new(c: Config) -> Result<Self> {
+        let config = aws_config::load_from_env().await;
+        let client = Client::new(&config);
+        Ok(Self { client, config: c })
+    }
+
+    /// Create an AWS KMS client using the default configutation.
+    pub async fn default() -> Result<Self> {
+        Self::new(Config::default()).await
+    }
+
+    /// Create a new NIST P-256 key-pair in AWS KMS and return its ID.
+    pub async fn create_key(&self) -> Result<KeyId> {
```

Comment:
can we change the `KeyId`  type here to be something else (maybe `KMSKeyId`  or something like that),  to make it clear that these are  *not*  the same thing as the `KeyId`  we index on?  (the ones that are a function of the public key).   IIUC  when signing for example we go from  
    `PubKey -->[hash_function()]-->  KeyId -->[lookup()]-->KMSKeyId -->[aws_kms_call(KMSKeyId)]--> signature`

Reply:
NB that `KeyId` is not a type but just an alias for `String`. This is why I felt I could reuse it here because it still is some sort of identifier of a key.

Reply:
Hum I see,   somehow I thought using a plain string where a KeyId was specified was not going to work,   but you are right is just an alias. 
I still feel it a bit confusing,   but can live with it.    Maybe in some future iteration we make the KeyId a newtype and so KMSKeyId ,   or  at least make another alias  with a different name    :man_shrugging: